Tasks/Maven/CodeAnalysis/mavenpmd.ts

```diff
@@ -28,6 +28,10 @@ export function collectPmdOutput(rootDir:string) : AnalysisResult {
 
     var pmdXmlFilePath = path.join(rootDir, 'target', 'pmd.xml');
     result = collectPmdXml(result, pmdXmlFilePath);
+    if (!result)
```

Comment:
Discussed offline. This way the code makes it clear that a null return from _collectPmdXML()_ results in a null return of its own.